### PR TITLE
Initialize featured and optional sections with correct data

### PR DESCRIPTION
**Pull request by bognix**
/cc @rogatty @vforge

### CODE REVIEW

File: front/scripts/main/models/CuratedContentEditorModel.ts [hunk 1]
```diff
@@ -93,14 +93,20 @@ App.CuratedContentEditorModel.reopenClass({
...
+				items: <any>[],
+				featured: 'true',
 			},
 			curated = {
 				items: <any>[]
 			},
 			optional = {
-				items: <any>[]
+				items: <any>[],
+				label: ''
```
Comment (vforge) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: Why do we need this? Optional section shouldn't have any label as far as I know.

Comment (rogatty) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: In multiple places we use empty string to identify optional section, for example in https://github.com/Wikia/app/blob/dev/extensions/wikia/CuratedContent/CuratedContentController.class.php#L347

```
// there also might be some categories without SECTION, lets find them as well (optional section)
$optionalSections = $this->getSectionItems( $content, '' );
```

File: front/scripts/main/models/CuratedContentEditorModel.ts [hunk 2]
```diff
@@ -93,14 +93,20 @@ App.CuratedContentEditorModel.reopenClass({
 	 * @returns {Object}
 	 */
 	sanitize(rawData: any): CuratedContentEditorModel {
+		/**
+		 * Label has to be initialized with string value because it can't be null.
```
Comment (rogatty) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: Seems too general, this is true only for "optional" block.

**Comment by rogatty** [comment]
```
front/scripts/main/models/CuratedContentEditorModel.ts(115,6): 2322 Type 'CuratedContentEditorRawSectionInterface' is not assignable to type '{ items: any; featured: string; }'.
  Property 'featured' is optional in type 'CuratedContentEditorRawSectionInterface' but required in type '{ items: any; featured: string; }'.
```

File: front/scripts/main/models/CuratedContentEditorModel.ts (hunk 1 above)
Comment (bognix) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: https://github.com/Wikia/app/blob/dev/extensions/wikia/CuratedContent/CuratedContentController.class.php#L357 and this particular line was the problematic one.

File: front/scripts/main/models/CuratedContentEditorModel.ts (hunk 2 above)
Comment (bognix) [inline front/scripts/main/models/CuratedContentEditorModel.ts]: true. At the beginning I thought it applies also to featured. I'll fix the comment.

**Comment by vforge** [comment]
LGTM

**Comment by hakubo** [comment]
LGTM

**Comment by rogatty** [comment]
LGTM

**Pull request merged**